File: stdlib.c
#include "stdio.h"

struct String {
    char str[256];
}; 

void print(struct String *s) {
    printf("%s\n", s->str);
}

struct String intToString(int a) {
    struct String s;
    sprintf(s.str, "%d", a);
    return s;
}

File: stdlib.hs
module StdLib where

import Grammar

stdLib :: [Stmt]
stdLib = [(Type "String" [])
        , (Func "pprint" [(Single "arg" "String")] "void" Nop)
        , (Func "intToString" [(Single "arg" "Int")] "String" Nop)
        , (Func "makeString" [(Array "s" "Char")] "void" Nop)
        ]

